src/NLog/Config/LoggingConfiguration.cs

```diff
@@ -330,6 +330,37 @@ public void AddRule(LogLevel minLevel, LogLevel maxLevel, Target target, string
             AddTargetThreadSafe(target.Name, target, false);
         }
 
+        /// <summary>
+        /// Add a rule with min- and maxLevel.
+        /// </summary>
+        /// <param name="ruleName">Name of the rule.</param>
+        /// <param name="minLevel">Minimum log level needed to trigger this rule.</param>
+        /// <param name="maxLevel">Maximum log level needed to trigger this rule.</param>
+        /// <param name="target">Target to be written to when the rule matches.</param>
+        /// <param name="loggerNamePattern">Logger name pattern. It may include the '*' wildcard at the beginning, at the end or at both ends.</param>
+        /// <param name="final">Gets or sets a value indicating whether to quit processing any further rule when this one matches.</param>
+        public void AddRule(string ruleName, LogLevel minLevel, LogLevel maxLevel, Target target, string loggerNamePattern, bool final)
```

Comment:
Think this overload should be removed, since `string ruleName` is an exotic feature. And normal users often mistake it with `string loggerNamePattern`.